server_src/lib/dataFunctions/fileClean.js

```diff
@@ -75,100 +75,83 @@ class FileClean {
-        try {
-          for (let f = 0; f < fileList.files.length; f++) {
-            const file = fileList.files[f];
-            const sortedFile = {
-              path: file.path,
-              fullPath: file.fullPath,
-              display: file.display,
-              name: file.name,
-              uploadDate: file.date,
-              fileSize: file.size,
-              thumbnail: file.thumbnail,
-              toolUnits: "",
-              toolCosts: "",
-              success: file.success,
-              failed: file.failed,
-              last: file.last,
-              expectedPrintTime: file.time,
-              printCost: getPrintCostNumeric(file.time, printCost),
-            };
-            sortedFile.toolUnits = await FileClean.getUnits(
-              selectedFilament,
-              file.length,
-            );
-            sortedFile.toolCosts = await FileClean.getCost(
-              selectedFilament,
-              sortedFile.toolUnits,
-            );
-            sortedFileList.push(sortedFile);
-            // console.log(sortedFileList);
-          }
-        } catch (e) {
-          console.log(e);
-        }
-      }
-      const newFileList = {
-        fileList: sortedFileList,
-        filecount: 0,
-        folderList: [],
-        folderCount: 0,
-      };
-      newFileList.filecount = fileList.fileCount;
-      newFileList.folderList = fileList.folders;
-      newFileList.folderCount = fileList.folderCount;
-      cleanFileList[farmPrinter.sortIndex] = newFileList;
-      if (typeof farmPrinter.systemChecks !== "undefined") {
-        farmPrinter.systemChecks.cleaning.file.status = "success";
-        farmPrinter.systemChecks.cleaning.file.date = new Date();
+    const { fileList } = farmPrinter;
+    if (!!farmPrinter.systemChecks) {
+      farmPrinter.systemChecks.cleaning.file.status = "warning";
+    }
+    const printCost = farmPrinter.costSettings;
+    const sortedFileList = [];
+    if (!!fileList) {
+      for (let file of fileList) {
+        const sortedFile = {
+          path: file.path,
+          fullPath: file.fullPath,
+          display: file.display,
+          name: file.name,
+          uploadDate: file.date,
+          fileSize: file.size,
+          thumbnail: file.thumbnail,
+          toolUnits: "",
+          toolCosts: "",
+          success: file.success,
+          failed: file.failed,
+          last: file.last,
+          expectedPrintTime: file.time,
+          printCost: getPrintCostNumeric(file.time, printCost),
+        };
+        sortedFile.toolUnits = FileClean.getUnits(
+          selectedFilament,
+          file.length,
+        );
+        sortedFile.toolCosts = FileClean.getCost(
+          selectedFilament,
+          sortedFile.toolUnits,
+        );
+        sortedFileList.push(sortedFile);
       }
-      logger.info("File Information cleaned and ready for consumption...");
-      return true;
-    } catch (e) {
-      console.log("MASSIVE FILE ERROR!!!!", e);
```

Comment:
Removed this, u ok with that?

Reply:
Yeah, is it caught somewhere if there are errors?

Reply:
What kind of errors do we foresee?

Reply:
We could validate the file property from file list. path, fullPath, display, name, date, size, success,failed, last, time. 

There's not really as much as I'd thought going on here when I made the comment.

Reply:
I started testing `statistics`, thats a bit more bug prone right now.

So in case an error occurs, I think saving a status variable and reason might help a lot in visualizing why certain cache's didnt generate. What do you think?